packages/php-wasm/universal/src/lib/wasm-error-reporting.ts

```diff
@@ -58,17 +55,26 @@ export function improveWASMErrorReporting(runtime: Runtime) {
 						e.cause = runtime.lastAsyncifyStackSource;
 					}
 
-					if (!target.hasListeners()) {
-						showCriticalErrorBox(clearMessage);
-						throw e;
+					if (target.hasListeners()) {
+						target.dispatchEvent(
+							new ErrorEvent('error', {
+								error: e,
+								message: clearMessage,
+							})
+						);
+						return;
 					}
 
-					target.dispatchEvent(
-						new ErrorEvent('error', {
-							error: e,
-							message: clearMessage,
-						})
-					);
+					const isExitCodeZero =
+						('exitCode' in e && e?.exitCode === 0) ||
+						(e?.name === 'ExitStatus' &&
+							'status' in e &&
+							e.status === 0);
+
+					if (!isExitCodeZero) {
+						showCriticalErrorBox(clearMessage);
+					}
+					throw e;
```

Comment:
The previous version of this PR skipped throwing, but turns out throwing is critical for Emscripten to correctly finish the execution and unwind Asyncify stack.